server/src/main/java/org/elasticsearch/search/rank/feature/RankFeatureResult.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search.rank.feature;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.RefCounted;
+import org.elasticsearch.core.SimpleRefCounted;
+import org.elasticsearch.search.SearchPhaseResult;
+import org.elasticsearch.search.SearchShardTarget;
+import org.elasticsearch.search.internal.ShardSearchContextId;
+import org.elasticsearch.search.internal.ShardSearchRequest;
+import org.elasticsearch.transport.LeakTracker;
+
+import java.io.IOException;
+
+/**
+ * The result of a rank feature search phase.
+ * Each instance holds a {@code RankFeatureShardResult} along with the references associated with it.
+ */
+public class RankFeatureResult extends SearchPhaseResult {
+
+    private RankFeatureShardResult rankShardResult;
+
+    private final RefCounted refCounted = LeakTracker.wrap(new SimpleRefCounted());
```

Comment:
Why have a ref counted here when we're not releasing anything once we get to `0` for the ref count?

Reply:
It was a leftover from a previous iteration; removed :)